File: src/api/java/BitVecSort.java
/**
Copyright (c) 2012-2014 Microsoft Corporation
   
Module Name:

    BitVecSort.java

Abstract:

Author:

    @author Christoph Wintersteiger (cwinter) 2012-03-15

Notes:
    
**/

package com.microsoft.z3;

/**
 * Bit-vector sorts.
 **/
public class BitVecSort extends Sort
{
    /**
     * The size of the bit-vector sort.
     * @throws Z3Exception on error
     * @return an int
     **/
    public int getSize()
    {
        return Native.getBvSortSize(getContext().nCtx(), getNativeObject());
    }

    BitVecSort(Context ctx, long obj)
    {
        super(ctx, obj);
    }
};

File: src/api/dotnet/BitVecSort.cs
/*++
Copyright (c) 2012 Microsoft Corporation

Module Name:

    BitVecSort.cs

Abstract:

    Z3 Managed API: BitVec Sorts

Author:

    Christoph Wintersteiger (cwinter) 2012-11-23

Notes:
    
--*/

using System.Diagnostics;
using System;

namespace Microsoft.Z3
{
    /// <summary>
    /// Bit-vector sorts.
    /// </summary>
    public class BitVecSort : Sort
    {
        /// <summary>
        /// The size of the bit-vector sort.
        /// </summary>
        public uint Size
        {
            get { return Native.Z3_get_bv_sort_size(Context.nCtx, NativeObject); }
        }

        #region Internal
        internal BitVecSort(Context ctx, IntPtr obj) : base(ctx, obj) { Debug.Assert(ctx != null); }
        #endregion
    };
}
